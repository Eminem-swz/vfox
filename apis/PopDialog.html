<!DOCTYPE html>
<html lang="zh-CN">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>PopDialog 气泡对话框 | Vfox for Vue3</title>
    <meta name="description" content="移动端 Vue 组件库">
    <link rel="stylesheet" href="/vfox/assets/style.82adce24.css">
    <link rel="modulepreload" href="/vfox/assets/Home.472bd3cf.js">
    <link rel="modulepreload" href="/vfox/assets/app.9dc78301.js">
    <link rel="modulepreload" href="/vfox/assets/apis_PopDialog.md.d4c2aefa.lean.js">
    
    <meta name="twitter:title" content="PopDialog 气泡对话框 | Vfox for Vue3">
  <meta property="og:title" content="PopDialog 气泡对话框 | Vfox for Vue3">
  </head>
  <body>
    <div id="app"><!--[--><div class="theme"><header class="nav-bar" data-v-675d8756><div class="sidebar-button" data-v-675d8756><svg class="icon" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z" class></path></svg></div><a class="nav-bar-title" href="/vfox/" aria-label="Vfox for Vue3, back to home" data-v-675d8756 data-v-4a583abe><!----> Vfox for Vue3</a><div class="flex-grow" data-v-675d8756></div><div class="nav" data-v-675d8756><nav class="nav-links" data-v-675d8756 data-v-eab3edfe><!--[--><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/" data-v-b8818f8c>指南 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/design/" data-v-b8818f8c>设计 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/components/Button" data-v-b8818f8c>组件 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item active" href="/vfox/apis/Toast" data-v-b8818f8c>API <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-dropdown-link" data-v-eab3edfe data-v-56bf3a3f><button class="button" data-v-56bf3a3f><span class="button-text" data-v-56bf3a3f>选择版本</span><span class="right button-arrow" data-v-56bf3a3f></span></button><ul class="dialog" data-v-56bf3a3f><!--[--><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item isExternal" href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>1.x</span><span class="icon" data-v-bbc27490><svg class="icon outbound" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewbox="0 0 100 100" width="15" height="15" data-v-bbc27490><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path><polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></span></a></div></li><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item active" href="/vfox/" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>2.x</span><span class="icon" data-v-bbc27490><!----></span></a></div></li><!--]--></ul></div></div><!--]--><!----><!----></nav></div><!--[--><!--]--></header><aside class="sidebar" data-v-83e92a68><nav class="nav-links nav" data-v-83e92a68 data-v-eab3edfe><!--[--><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/" data-v-b8818f8c>指南 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/design/" data-v-b8818f8c>设计 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/components/Button" data-v-b8818f8c>组件 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item active" href="/vfox/apis/Toast" data-v-b8818f8c>API <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-dropdown-link" data-v-eab3edfe data-v-56bf3a3f><button class="button" data-v-56bf3a3f><span class="button-text" data-v-56bf3a3f>选择版本</span><span class="right button-arrow" data-v-56bf3a3f></span></button><ul class="dialog" data-v-56bf3a3f><!--[--><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item isExternal" href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>1.x</span><span class="icon" data-v-bbc27490><svg class="icon outbound" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewbox="0 0 100 100" width="15" height="15" data-v-bbc27490><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path><polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></span></a></div></li><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item active" href="/vfox/" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>2.x</span><span class="icon" data-v-bbc27490><!----></span></a></div></li><!--]--></ul></div></div><!--]--><!----><!----></nav><!--[--><!--]--><ul class="sidebar-links" data-v-83e92a68><!--[--><li class="sidebar-link"><p class="sidebar-link-item">通知类弹窗</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Toast">Toast</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Notify">Notify</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">预览类弹窗</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Preview">Preview</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">模态框类弹窗</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Dialog">Dialog</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">抽屉类弹窗</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/ActionSheet">ActionSheet</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Picker">Picker</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/DatePicker">DatePicker</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Cascader">Cascader</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Calendar">Calendar</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">气泡类弹窗</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Popover">Popover</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item active" href="/vfox/apis/PopDialog">PopDialog</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#showpopdialog-object">showPopDialog(object)</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#object">object</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#usage">Usage</a><!----></li></ul></li></ul></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/PopMenu">PopMenu</a><!----></li></ul></li><!--]--></ul><!--[--><!--]--></aside><div class="sidebar-mask"></div><main class="page" data-v-7eddb2c4><div class="container" data-v-7eddb2c4><!--[--><!--[--><!----><!--]--><!--[--><div><iframe style="display:none;width:375px;height:667px;margin-top:1.5rem;" src="https://godxiaoji.github.io/vfox/demo/#/" frameborder="0"></iframe></div><!--]--><!--]--><div style="position:relative;" class="content" data-v-7eddb2c4><div><h1 id="popdialog-气泡对话框" tabindex="-1">PopDialog 气泡对话框 <a class="header-anchor" href="#popdialog-气泡对话框" aria-hidden="true">#</a></h1><h2 id="showpopdialog-object" tabindex="-1">showPopDialog(object) <a class="header-anchor" href="#showpopdialog-object" aria-hidden="true">#</a></h2><p>显示气泡对话框。</p><h3 id="object" tabindex="-1">object <a class="header-anchor" href="#object" aria-hidden="true">#</a></h3><table><thead><tr><th>属性</th><th>类型</th><th>默认值</th><th>必填</th><th>说明</th></tr></thead><tbody><tr><td>selector</td><td>string | HTMLElement</td><td></td><td>是</td><td>从哪个元素展开，如果是 string，则为可以被 document.querySelector(selector) 获取到</td></tr><tr><td>placement</td><td>string</td><td>&#39;bottom&#39;</td><td>否</td><td>展开位置，可选 &#39;bottom&#39;, &#39;top&#39;, &#39;left&#39;, &#39;right&#39;</td></tr><tr><td>content</td><td>string</td><td></td><td>是</td><td>文本内容</td></tr><tr><td>showCancel</td><td>boolean</td><td>true</td><td>否</td><td>是否显示取消按钮</td></tr><tr><td>cancelText</td><td>string</td><td>&#39;取消&#39;</td><td>否</td><td>取消按钮的文本</td></tr><tr><td>confirmText</td><td>string</td><td>&#39;确定&#39;</td><td>否</td><td>确定按钮的文本</td></tr><tr><td>success</td><td>function</td><td></td><td>否</td><td>接口调用成功（在用户做出选择后，如取消，确定）的回调函数</td></tr><tr><td>fail</td><td>function</td><td></td><td>否</td><td>接口调用失败（如传入错误的参数）的回调函数（不传入 fail 遇错误直接抛出）</td></tr><tr><td>complete</td><td>function</td><td></td><td>否</td><td>弹窗关闭或调用失败的回调函数</td></tr></tbody></table><h4 id="object-success-回调参数" tabindex="-1">object.success 回调参数 <a class="header-anchor" href="#object-success-回调参数" aria-hidden="true">#</a></h4><table><thead><tr><th>属性</th><th>类型</th><th>说明</th></tr></thead><tbody><tr><td>cancel?</td><td>boolean</td><td>为 true 时，表示取消</td></tr><tr><td>confirm?</td><td>boolean</td><td>为 true 时，表示点击了确定</td></tr></tbody></table><h3 id="usage" tabindex="-1">Usage <a class="header-anchor" href="#usage" aria-hidden="true">#</a></h3><p>具体调用方式可以参考<a href="./../guide/import.html#api-调用">API 调用</a>。</p><div class="language-JavaScript"><pre><code><span class="token function">showPopDialog</span><span class="token punctuation">(</span><span class="token punctuation">{</span>
  <span class="token literal-property property">selector</span><span class="token operator">:</span> <span class="token string">&#39;#popDialogTarget&#39;</span><span class="token punctuation">,</span>
  <span class="token literal-property property">content</span><span class="token operator">:</span> <span class="token keyword">this</span><span class="token punctuation">.</span>content<span class="token punctuation">,</span>
  <span class="token function-variable function">success</span><span class="token operator">:</span> <span class="token punctuation">(</span><span class="token parameter">res</span><span class="token punctuation">)</span> <span class="token operator">=&gt;</span> <span class="token punctuation">{</span>
    console<span class="token punctuation">.</span><span class="token function">log</span><span class="token punctuation">(</span><span class="token string">&#39;success&#39;</span><span class="token punctuation">,</span> res<span class="token punctuation">)</span>
  <span class="token punctuation">}</span>
<span class="token punctuation">}</span><span class="token punctuation">)</span>
</code></pre></div></div></div><footer class="page-footer" data-v-7eddb2c4 data-v-fb8d84c6><div class="edit" data-v-fb8d84c6><div class="edit-link" data-v-fb8d84c6 data-v-1ed99556><!----></div></div><div class="updated" data-v-fb8d84c6><!----></div></footer><div class="next-and-prev-link" data-v-7eddb2c4 data-v-38ede35f><div class="container" data-v-38ede35f><div class="prev" data-v-38ede35f><a class="link" href="/vfox/apis/Popover" data-v-38ede35f><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" class="icon icon-prev" data-v-38ede35f><path d="M19,11H7.4l5.3-5.3c0.4-0.4,0.4-1,0-1.4s-1-0.4-1.4,0l-7,7c-0.1,0.1-0.2,0.2-0.2,0.3c-0.1,0.2-0.1,0.5,0,0.8c0.1,0.1,0.1,0.2,0.2,0.3l7,7c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3c0.4-0.4,0.4-1,0-1.4L7.4,13H19c0.6,0,1-0.4,1-1S19.6,11,19,11z"></path></svg><span class="text" data-v-38ede35f>Popover</span></a></div><div class="next" data-v-38ede35f><a class="link" href="/vfox/apis/PopMenu" data-v-38ede35f><span class="text" data-v-38ede35f>PopMenu</span><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" class="icon icon-next" data-v-38ede35f><path d="M19.9,12.4c0.1-0.2,0.1-0.5,0-0.8c-0.1-0.1-0.1-0.2-0.2-0.3l-7-7c-0.4-0.4-1-0.4-1.4,0s-0.4,1,0,1.4l5.3,5.3H5c-0.6,0-1,0.4-1,1s0.4,1,1,1h11.6l-5.3,5.3c-0.4,0.4-0.4,1,0,1.4c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3l7-7C19.8,12.6,19.9,12.5,19.9,12.4z"></path></svg></a></div></div></div><!--[--><!--]--></div></main></div><!----><!--]--></div>
    <script>__VP_HASH_MAP__ = JSON.parse("{\"apis_actionsheet.md\":\"6e759880\",\"apis_calendar.md\":\"73c56524\",\"apis_cascader.md\":\"6de70566\",\"apis_datepicker.md\":\"e6503750\",\"apis_dialog.md\":\"6964068d\",\"apis_notify.md\":\"9d557c5d\",\"apis_picker.md\":\"6ad5b8f1\",\"apis_popdialog.md\":\"d4c2aefa\",\"apis_popmenu.md\":\"a0c9cfbe\",\"apis_popover.md\":\"a3645836\",\"apis_preview.md\":\"8b35e969\",\"apis_toast.md\":\"7568137d\",\"apis_index.md\":\"b1f704ad\",\"components_actionsheet.md\":\"52d22cba\",\"components_activityindicator.md\":\"b8a223e7\",\"components_avatar.md\":\"011d3fc7\",\"components_backtop.md\":\"e4e994f5\",\"components_badge.md\":\"028a7d96\",\"components_button.md\":\"8bf05d38\",\"components_calendar.md\":\"a1d6715d\",\"components_cascader.md\":\"66e4f298\",\"components_cell.md\":\"4ea2f836\",\"components_checkbox.md\":\"47b2c132\",\"components_circleprogress.md\":\"e5fcacc5\",\"components_collapse.md\":\"b43c15e1\",\"components_copy.md\":\"9e2aafef\",\"components_countdown.md\":\"1269c173\",\"components_countup.md\":\"f9854800\",\"components_datepicker.md\":\"87f0a811\",\"components_dialog.md\":\"25aab6d4\",\"components_divider.md\":\"d273a4eb\",\"components_drawer.md\":\"95ea9b04\",\"components_dropdown.md\":\"af86545f\",\"components_empty.md\":\"ca6616bb\",\"components_fixed.md\":\"31e38fdf\",\"components_flatlist.md\":\"f7a97393\",\"components_form.md\":\"3e13777e\",\"components_group.md\":\"ba614409\",\"components_icon.md\":\"c13bdbb3\",\"components_image.md\":\"cccd71f1\",\"components_imagepreview.md\":\"0139cb97\",\"components_imageuploader.md\":\"2c5b4d8c\",\"components_indexview.md\":\"b8b8cfb8\",\"components_input.md\":\"f8e72d81\",\"components_layout.md\":\"cabaf94c\",\"components_loadmore.md\":\"06109345\",\"components_modal.md\":\"872e1dd2\",\"components_navbar.md\":\"4750069d\",\"components_noticebar.md\":\"7d2b7c72\",\"components_notify.md\":\"b9eab969\",\"components_numberkeyboard.md\":\"896a2a73\",\"components_order.md\":\"4e95c01a\",\"components_pagination.md\":\"e4f0475d\",\"components_picker.md\":\"7af0057b\",\"components_popdialog.md\":\"d934a048\",\"components_popmenu.md\":\"2ce4cf62\",\"components_popover.md\":\"5ca63491\",\"components_price.md\":\"9799d29b\",\"components_progress.md\":\"b28c8651\",\"components_radio.md\":\"421c7a10\",\"components_range.md\":\"727c91aa\",\"components_rate.md\":\"a065896e\",\"components_result.md\":\"30cd9913\",\"components_scrolltab.md\":\"e12e16c3\",\"components_scrollview.md\":\"950370c9\",\"components_searchbar.md\":\"856dffb5\",\"components_sidetab.md\":\"ee102926\",\"components_skeleton.md\":\"821db3fc\",\"components_slider.md\":\"d7006cc7\",\"components_stepper.md\":\"49f3b7c2\",\"components_steps.md\":\"877b5ea0\",\"components_sticky.md\":\"16f4fc4c\",\"components_stopwatch.md\":\"e3735e07\",\"components_swipecell.md\":\"4eb6b4bb\",\"components_swiper.md\":\"586c4b1f\",\"components_switch.md\":\"9c810987\",\"components_tab.md\":\"40fea87a\",\"components_tabbar.md\":\"6641be22\",\"components_tabview.md\":\"5ddcafa6\",\"components_tag.md\":\"9e1e77f3\",\"components_timeago.md\":\"f342d517\",\"components_timeline.md\":\"0b56084b\",\"components_toast.md\":\"b942008b\",\"components_index.md\":\"8bc2015a\",\"design_color.md\":\"b30ef7ca\",\"design_index.md\":\"7ede16dc\",\"guide_dark.md\":\"01b89ba9\",\"guide_import.md\":\"6af40c44\",\"guide_locale.md\":\"a67725ac\",\"guide_theme.md\":\"a168b5c6\",\"index.md\":\"d0d3ac39\"}")</script>
    <script type="module" async src="/vfox/assets/app.9dc78301.js"></script>
    
  </body>
</html>