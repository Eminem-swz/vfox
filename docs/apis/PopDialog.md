# PopDialog 气泡对话框

## showPopDialog(object)

显示气泡对话框。

### object

| 属性        | 类型                  | 默认值   | 必填 | 说明                                                                              |
| ----------- | --------------------- | -------- | ---- | --------------------------------------------------------------------------------- |
| selector    | string \| HTMLElement |          | 是   | 从哪个元素展开，如果是 string，则为可以被 document.querySelector(selector) 获取到 |
| placement   | string                | 'bottom' | 否   | 展开位置，可选 'bottom', 'top', 'left', 'right'                                   |
| content     | string                |          | 是   | 文本内容                                                                          |
| showCancel  | boolean               | true     | 否   | 是否显示取消按钮                                                                  |
| cancelText  | string                | '取消'   | 否   | 取消按钮的文本                                                                    |
| confirmText | string                | '确定'   | 否   | 确定按钮的文本                                                                    |
| success     | function              |          | 否   | 接口调用成功（在用户做出选择后，如取消，确定）的回调函数                          |
| fail        | function              |          | 否   | 接口调用失败（如传入错误的参数）的回调函数（不传入 fail 遇错误直接抛出）          |
| complete    | function              |          | 否   | 弹窗关闭或调用失败的回调函数                                                      |

#### object.success 回调参数

| 属性     | 类型    | 说明                       |
| -------- | ------- | -------------------------- |
| cancel?  | boolean | 为 true 时，表示取消       |
| confirm? | boolean | 为 true 时，表示点击了确定 |

### Usage

具体调用方式可以参考[API 调用](../guide/import.md#api-调用)。

```JavaScript
showPopDialog({
  selector: '#popDialogTarget',
  content: this.content,
  success: (res) => {
    console.log('success', res)
  }
})
```
