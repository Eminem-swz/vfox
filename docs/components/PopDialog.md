# PopDialog 气泡对话框

## Import

```JavaScript
import { PopDialog } from 'vfox'
```

具体的引入方式可以参考[引入组件](../guide/import.md)。

## Props

| 属性            | 类型                  | 默认值   | 必填 | 说明                                                                              |
| --------------- | --------------------- | -------- | ---- | --------------------------------------------------------------------------------- |
| v-model:visible | boolean               | false    | 否   | 是否显示                                                                          |
| selector        | string \| HTMLElement |          | 是   | 从哪个元素展开，如果是 string，则为可以被 document.querySelector(selector) 获取到 |
| placement       | PlacementType         | 'bottom' | 否   | 展开位置，可选 'bottom', 'top', 'left', 'right'                                   |
| show-mask       | boolean               | true     | 否   | 是否展示蒙层，如果设置不展示，气泡则是跟随 `selector` 对应的元素                  |
| content         | string                |          | 是   | 文本内容                                                                          |
| show-cancel     | boolean               | true     | 否   | 是否显示取消按钮                                                                  |
| cancel-text     | string                | '取消'   | 否   | 取消按钮的文本                                                                    |
| confirm-text    | string                | '确定'   | 否   | 确定按钮的文本                                                                    |

## Events

| 事件                 | 描述               | 回调函数参数                                                           | TypeScript 函数           |
| -------------------- | ------------------ | ---------------------------------------------------------------------- | ------------------------- |
| cancel               | 取消按钮点击时触发 | payload: { source: string }                                            | PopupOnCancel             |
| confirm              | 确认按钮点击时触发 |                                                                        |
| visible-state-change | 展示隐藏时触发     | payload: { state: [VisibleState](./PopDialog.md#visiblestate-值说明) } | PopupOnVisibleStateChange |

### VisibleState 值说明

| 值     | 说明                 | 备注                                              |
| ------ | -------------------- | ------------------------------------------------- |
| show   | 展示时触发           |                                                   |
| shown  | 展示且动画结束后触发 |                                                   |
| hide   | 隐藏时触发           | 可能携带其他参数 cancel, maskClick, closeClick 等 |
| hidden | 隐藏且动画结束后触发 | 可能携带其他参数 cancel, maskClick, closeClick 等 |
