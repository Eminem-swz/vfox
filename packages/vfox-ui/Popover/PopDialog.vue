<template>
  <teleport to="body">
    <div
      class="fx-popover fx-popup fx-pop-dialog"
      :class="{ visible: visible2, 'no--mask': !showMask }"
      :style="popupStyles"
      v-bind="$attrs"
      v-show="isShow"
    >
      <div class="fx-mask" @click="onMaskClick"></div>
      <div class="fx-popover_inner" ref="innerEl" :style="innerStyles">
        <i class="fx-popover_arrow" :style="arrowStyles"></i>
        <div class="fx-popover_content">
          <div class="fx-pop-dialog_body">
            <div class="fx-popover_text">{{ content }}</div>
          </div>
          <div class="fx-pop-dialog_footer fx-horizontal-hairline">
            <ButtonGroup
              class="fx-pop-dialog_footer-inner"
              pattern="borderless"
              size="middle"
            >
              <Button v-if="showCancel" type="default" @click="onCancelClick">
                {{ cancelText || locale.popDialogCancelText }}
              </Button>
              <Button type="primary" @click="onConfirmClick">
                {{ confirmText || locale.popDialogConfirmText }}
              </Button>
            </ButtonGroup>
          </div>
        </div>
      </div>
    </div>
  </teleport>
</template>

<script lang="ts">
import { defineComponent } from 'vue'
import { usePopover } from '../Popover/use-popover'
import { popoverProps, popoverEmits } from '../Popover/popover'
import { Button, ButtonGroup } from '../Button'
import { locale } from '../Locale'

export default defineComponent({
  name: 'fx-pop-dialog',
  components: { Button, ButtonGroup },
  props: {
    ...popoverProps,
    content: {
      type: String,
      default: '',
      required: true
    },
    // 展示取消按钮
    showCancel: {
      type: Boolean,
      default: true
    },
    // 取消文本
    cancelText: {
      type: String
    },
    // 确认文本
    confirmText: {
      type: String
    }
  },
  emits: { ...popoverEmits },
  setup(props, ctx) {
    const popoverHook = usePopover(props, ctx)

    function onConfirmClick() {
      popoverHook.customConfirm({})
    }

    return {
      ...popoverHook,
      onConfirmClick,
      locale
    }
  }
})
</script>
