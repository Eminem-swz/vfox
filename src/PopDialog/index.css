.fx-pop-dialog_footer {
  display: flex;
  flex-direction: column;
}

.fx-pop-dialog_footer::before {
  content: '';
}

.fx-pop-dialog_footer-inner {
  display: flex;
  justify-content: space-between;
  position: relative;
}

.fx-pop-dialog_footer-inner > .fx-button + .fx-button::before {
  height: 100%;
  background-color: #d9d9d9;
}
