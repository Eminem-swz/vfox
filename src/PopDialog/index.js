import PopDialog from '../Popover/Dialog.vue'
import { showPopDialog } from '../apis/Popup'

PopDialog.install = function(app) {
  app.component(PopDialog.name, PopDialog)

  // 添加api
  app.config.globalProperties.$showPopDialog = showPopDialog
}

PopDialog.showPopDialog = showPopDialog

export default PopDialog
