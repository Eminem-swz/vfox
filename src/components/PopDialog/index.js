import PopDialog from '../Popover/Dialog.vue'

PopDialog.install = function(Vue) {
  Vue.component(PopDialog.name, PopDialog)
}

export default PopDialog
