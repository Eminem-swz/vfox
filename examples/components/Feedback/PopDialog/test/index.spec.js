/* global describe,test,expect */

import { mount } from '@vue/test-utils'
import PopDialog from '@/components/PopDialog'

describe('PopDialog', () => {
  test('should render default correctly', () => {
    const wrapper = mount(PopDialog, {
      propsData: {
        selector: '#test',
        title: 'title',
        content: 'content'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should render prop showCancel=false correctly', () => {
    const wrapper = mount(PopDialog, {
      propsData: {
        selector: '#test',
        title: 'title',
        content: 'content',
        showCancel: false
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
