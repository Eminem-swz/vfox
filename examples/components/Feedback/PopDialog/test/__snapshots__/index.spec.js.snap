// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopDialog should render default correctly 1`] = `
<!--teleport start-->
<!--teleport end-->
`;

exports[`PopDialog should render prop showCancel=false correctly 1`] = `
<!--teleport start-->
<!--teleport end-->
`;
